awx/main/credential_plugins/aim.py

```diff
@@ -0,0 +1,83 @@
+from .plugin import CredentialPlugin
+
+import os
+import stat
+import tempfile
+import threading
+from urllib.parse import urljoin
+
+from django.utils.translation import ugettext_lazy as _
+import requests
+
+
+aim_inputs = {
+    'fields': [{
+        'id': 'url',
+        'label': _('AIM URL'),
+        'type': 'string',
+    }, {
+        'id': 'application_id',
+        'label': _('Application ID'),
+        'type': 'string',
+        'secret': True,
+    }, {
+        'id': 'client_key',
+        'label': _('Client Key'),
+        'type': 'string',
+        'secret': True,
+        'multiline': True,
+    }, {
+        'id': 'client_cert',
+        'label': _('Client Cert'),
+        'type': 'string',
+        'secret': True,
+        'multiline': True,  
+    }],
+    'metadata': [{
+        'id': 'safe',
+        'label': _('Safe'),
+        'type': 'string',
+    }, {
+        'id': 'object',
+        'label': _('Object'),
+        'type': 'string',
+    }],
+    'required': ['url', 'application_id', 'safe', 'object'],
+}
+
+def temp_fifo(data):
+    # Open fifo named pipe in a new thread.
+    # The thread blocks until data is read from the pipe.
+    path = os.path.join(tempfile.mkdtemp(), next(tempfile._get_candidate_names()))
+    os.mkfifo(path, stat.S_IRUSR | stat.S_IWUSR)
+    threading.Thread(
+        target=lambda p, d: open(p, 'wb').write(d),
+        args=(path, data.encode())
+    ).start()
+    return path
+
+def aim_backend(**kwargs):
+    url = kwargs['url']
+    client_cert = kwargs.get('client_cert', None)
+    client_key = kwargs.get('client_key', None)
+
+    request_qs = '?AppId={application_id}&Safe={safe}&object={object}'.format(**kwargs)
+    request_url = urljoin(url, '/'.join(['AIMWebService', 'api', 'Accounts']))
+
+    client_cert_path = temp_fifo(client_cert)
+
+    cert = client_cert_path
+    if client_cert and client_key:
+        client_key_path = temp_fifo(client_key)
+        cert = (client_cert_path, client_key_path)
+
+    res = requests.get(request_url + request_qs, cert=cert, verify=False)
```

Comment:
`verify=False` probably isn't a secure default to assume

Reply:
added as a param to the credential, with default set to True